doc/upgrade-cluster.md

```diff
@@ -58,6 +58,36 @@ To initiate an upgrade process, a user has to update the `spec.version` field on
 
 After the `version` field is updated, the operator will detect the version change and it will trigger the upgrade process.
 
+### Upgrade guide
+
+> Note: To trigger an upgrade please edit only the fields mentioned in this upgrade guide. We do not recommend clubbing other edits (for performance or scaling or anything else) along with the upgrade trigger. Those can be done either prior to triggering the upgrade or after the upgrade has completed successfully.
+
+#### Upgrades till Pravega 0.6
+
+When upgrading the Pravega Cluster to any version till 0.6, we need to ensure that we do not modify any field other than the `version` field while triggering the upgrade. Altering of any other parameter can be handled either before triggering the upgrade or after the upgrade has successfully completed.
+
+#### Upgrade to Pravega 0.7 or above from a version below 0.7
+
+When upgrading the Pravega Cluster from any version below 0.7 to version 0.7 or above, there are a few configuration changes that must be made to Pravega manifest either with the upgrade request or prior to starting the upgrade. This is because from version 0.7 onwards, we no longer use Rocks DB for the Pravega Segmenstore. These configuration changes are needed to ensure that we have setup memory and other parameters appropriately for segmentstore streaming cache to work as expected.
+
+The additional changes that need to be made to the Pravega manifest are the following
+
+1. Ensure that the **memory limits** field within **Segmentstore Resources** is set to an appropriate value (let's call it the POD_MEM_LIMIT). The Segment Store cannot consume an amount of memory higher than this, otherwise we will get an Out Of Memory Killed error and the pod will be evicted.
+
+2. We need to distribute the pod's memory (M) between JVM Heap and Direct Memory. For instance, if POD_MEM_LIMIT=16GB then we can set 4GB for JVM and the rest for Direct Memory (12GB) i.e. POD_MEM_LIMIT (16GB) = JVM Heap (4GB) + Direct Memory (12GB).
```

Comment:
Remove the "We need to" you can start with "Distribute ..." Or say "It is recommended to distribute..."